powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/util/EciesUtils.java

```diff
@@ -54,22 +60,96 @@ public static long generateTimestamp() {
 
     /**
      * Derive associated data for ECIES.
-     * @param eciesScope ECIES scope.
-     * @param version Protocol version.
+     * @param scope Encryptor's scope.
+     * @param protocolVersion Protocol version.
      * @param applicationKey Application key.
      * @param activationId Activation ID.
      * @return Derived associated data.
+     * @throws EciesException In case that activation ID is required but is missing.
      */
-    public static byte[] deriveAssociatedData(EciesScope eciesScope, String version, String applicationKey, String activationId) {
-        if ("3.2".equals(version)) {
-            if (eciesScope == EciesScope.ACTIVATION_SCOPE) {
-                return ByteUtils.concatStrings(version, applicationKey, activationId);
+    public static byte[] deriveAssociatedData(EncryptorScope scope, String protocolVersion, String applicationKey, String activationId) throws EciesException {
+        if ("3.2".equals(protocolVersion)) {
+            if (protocolVersion == null) {
```

Comment:
This can never happen, the `protocolPersion` variable is `"3.3"` according to the condition just above... the condition should probably be above?